src/hotspot/share/gc/z/zVirtualMemory.cpp

```diff
@@ -99,16 +98,56 @@ size_t ZVirtualMemoryManager::reserve_discontiguous(size_t size) {
   return reserved;
 }
 
+bool ZVirtualMemoryManager::reserve_contiguous_inner(uintptr_t start, size_t size) {
+  assert(is_aligned(size, ZGranuleSize), "Must be granule aligned");
+
+  // Reserve address views
+  const uintptr_t marked0 = ZAddress::marked0(start);
+  const uintptr_t marked1 = ZAddress::marked1(start);
+  const uintptr_t remapped = ZAddress::remapped(start);
+
+  // Reserve address space
+  if (os_reserve(marked0, size) != marked0) {
+    return false;
+  }
+
+  if (os_reserve(marked1, size) != marked1) {
+    os_unreserve(marked0, size);
+    return false;
+  }
+
+  if (os_reserve(remapped, size) != remapped) {
+    os_unreserve(marked0, size);
+    os_unreserve(marked1, size);
+    return false;
+  }
+
+  // Register address views with native memory tracker
+  nmt_reserve(ZAddress::marked0(start), size);
```

Comment:
The first arguments are the same as locals `marked0`, `marked1`, `remapped` above, right? Can be reused then?

Reply:
Yes. At one point I had this part extracted into a separate function and had to re-resolve those values.